worldwind/src/commonMain/kotlin/earth/worldwind/draw/DrawShapeState.kt

```diff
@@ -5,16 +5,78 @@ import earth.worldwind.geom.Vec3
 import earth.worldwind.render.Color
 import earth.worldwind.render.Texture
 import earth.worldwind.render.buffer.AbstractBufferObject
-import earth.worldwind.render.buffer.FloatBufferObject
 import earth.worldwind.render.program.TriangleShaderProgram
+import earth.worldwind.util.kgl.GL_FLOAT
+
+open class VertexState {
+    internal class VertexAttrib constructor(
+        var index: Int = 0,
+        var vertexBuffer: AbstractBufferObject? = null,
+        var size: Int = 0,
+        var type: Int = GL_FLOAT,
+        var normalized: Boolean = false,
+        var stride: Int = 0,
+        var offset: Int = 0
+    ) {
+    }
+
+    private var attributes = mutableListOf<VertexAttrib>()
+
+    fun reset() {
+        attributes.clear()
+    }
+
+    fun addAttribute(
+        index: Int,
+        vertexBuffer: AbstractBufferObject,
+        size: Int,
+        type: Int,
+        normalized: Boolean,
+        stride: Int,
+        offset: Int
+    ) {
+        attributes.add(VertexAttrib(index, vertexBuffer, size, type, normalized, stride, offset))
+    }
+
+    fun bind(dc: DrawContext): Boolean {
+        var bindSuccessful = true
+        for (vertexAttrib in attributes) {
+            if (vertexAttrib.vertexBuffer != null) {
+                bindSuccessful = vertexAttrib.vertexBuffer!!.bindBuffer(dc)
```

Comment:
Do not bind on very iteration only if currentky bind does not match

Reply:
DrawContext saves last bound buffer and checks if we try to bind it again, should we duplicate this logic here?

Reply:
No, then just ensure there are only non null buffers in interface and skip `!!` check as it can be very expensive

Reply:
done